doc/hardware/porting/board_porting.rst

```diff
@@ -65,112 +65,128 @@ complete conversion reference.
+   | :ref:`mimx8mp_evk <imx8mp_evk>`            | mimx8ml8/a53             | i.MX8M Plus | i.MX8M             | NXP i.MX           | Arm Cortex-A53 | ARMv8-A              |
+   |                                            +--------------------------+-------------+--------------------+--------------------+----------------+----------------------+
+   |                                            | mimx8ml8/m7              | i.MX8M Plus | i.MX8M             | NXP i.MX           | Arm Cortex-M7  | ARMv7-M              |
+   +--------------------------------------------+--------------------------+-------------+--------------------+--------------------+----------------+----------------------+
+
+Additional details about terminology can be found in the next section.
+
+.. _board_terminology:
+
+Board terminology
+*****************
+
+The previous section introduced the hierarchical manner in which Zephyr
+classifies and implements hardware support.
+This section focuses on the terminology used around hardware support, and in
+particular when defining and working with boards and SoCs.
+
+The overall set of terms used around the concept of board in Zephyr is depicted
+in the image below, which uses the :ref:`bl5340_dvk` board as reference.
+
+.. figure:: board/board-terminology.svg
+   :width: 500px
+   :align: center
+   :alt: Board terminology diagram
 
-So to build hello world for ``plank``, variant ``foo``, you can do:
+   Board terminology diagram
 
-.. code-block:: console
+The diagram shows the different terms that are used to describe boards:
 
-   west build -b plank//foo samples/hello_world
+- The :term:`board name`: ``bl5340_dvk``
+- The optional :term:`board revision`: ``1.2.0``
+- The :term:`board qualifiers`, that optionally describe the :term:`SoC`,
+  :term:`CPU cluster` and :term:`variant`: ``nrf5340/cpuapp/ns``
+- The :term:`board target`, which uniquely identifies a combination of the above
+  and can be used to specify the hardware to build for when using the tooling
+  provided by Zephyr: ``bl5340_dvk@1.2.0/nrf5340/cpuapp/ns``
 
-When using multi-core SoCs, the CPU cluster is identified after the SoC
-identifier.
+Formally this can also be seen as:
 
-If ``soc1`` above has two cores, ``first`` and ``second``, then those are
-identified as: ``plank/soc1/first`` and ``plank/soc1/second``.
+.. code-block:: console
 
-And similar to before, if the board has only a single SoC, the SoC can be
-omitted, that is ``plank//first`` and ``plank//second`` is an identical short
-form.
+   {board name}[@{revision}][/{board qualifiers}]
 
-.. _hw_support_hierarchy:
+which can be extended to:
 
-Boards, SoCs, etc.
-******************
+.. code-block:: console
 
-Zephyr's hardware support hierarchy has these layers, from most to least
-specific:
+   {board name}[@{revision}][/{SoC}[/{CPU cluster}][/{variant}]]
```

Comment:
The combination of `[ ]`s and `{ }`s makes this really hard to parse visually IMO
Please consider the following:

```
Formally this can also be seen as
:samp:`{board name}[@{revision}][/{board qualifiers}]`, which can be extended to
:samp:`{board name}[@{revision}][/{SoC}[/{CPU cluster}][/{variant}]]`.
```

<img width="854" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/128251/ff9ffcb5-36d2-40cc-8a73-e488b454b7cc">